conda_index/index/__init__.py

```diff
@@ -901,41 +892,33 @@ def _append_group(groups, candidate):
         groups = []
         package_groups = groupby(lambda x: x[1]["name"], all_repodata_packages.items())
         for groupname, group in package_groups.items():
-            if groupname not in package_data or package_data[groupname].get(
```

Comment:
This was the important change. We "pay special attention" to everything now, so that we don't get older-version-newer-timestamp in a channeldata.json-from-scratch, compared to a channeldata.json-not-from-scratch.